sw/device/lib/pinmux.c

```diff
@@ -11,21 +11,15 @@
 
 #define PINMUX0_BASE_ADDR TOP_EARLGREY_PINMUX_AON_BASE_ADDR
 
-// TODO: autogenerated constants for this will be available once
-// the pinmux datastructure has been overhauled.
 #define NUM_GPIO 32
-#define UART_RX_PERIPH_IDX 32
-#define UART_TX_PERIPH_IDX 32
-#define MIO_32_IOR2_IDX 32
-#define MIO_33_IOR3_IDX 33
 
-static void init_pinmux_reg(uint32_t reg, uint32_t mask, uint32_t start_v) {
+static void init_gpio_regs(uint32_t reg, uint32_t mask, uint32_t start_v) {
   mmio_region_t reg32 = mmio_region_from_addr(reg);
   uint32_t reg_value = start_v;
-  uint32_t reg_offset = 0;
+  uint32_t reg_offset = kTopEarlgreyPinmuxPeripheralInGpioGpio0 << 2;
```

Comment:
We need to convert the enumeration index into a byte address.
As each register contains 4 bytes -> `<< 2`. I'll add a comment.

Reply:
it might be good to make this a parameter inside pinmux_reg_pkg that software can directly reference. 
if we change in the future, then this would automatically be covered.